R/as_point.R

```diff
@@ -0,0 +1,109 @@
+#' @title Convert a forecast object to a point forecast
+#'
+#' @description
+#' The `as_point` function is used to convert a forecast object to a point
+#' forecast. It is a generic function that dispatches the conversion to the
+#' appropriate method based on the class of the input forecast object.
+#'
+#' @param forecast An object of class `forecast_{type}`,
+#' representing a forecast.
+#' @param ... Additional arguments to be passed to the specific method.
+#' @return The function returns a point forecast object, which is a specific
+#' type of forecast object that represents a single value prediction.
+#'
+#' @export
+#' @keywords check-forecasts
+#' @examples
+#' as_point(as_forecast(example_quantile))
+#' as_point(as_forecast(example_sample_continuous))
+as_point <- function(forecast, ...) {
+  UseMethod("as_point")
+}
+
+#' @rdname as_point
+#' @export
+#' @importFrom cli cli_abort
+as_point.default <- function(forecast, ...) {
+  cli_abort(
+    c(
+      "!" = "The input needs to be a forecast object.",
+      "i" = "Please run `as_forecast()` first." # nolint
+    )
+  )
+}
+
+#' Convert a quantile forecast to a point forecast
+#'
+#' This function takes a quantile forecast and converts it to a point forecast
+#' by selecting the forecast corresponding to the specified quantile level.
+#'
+#' @param forecast The `forecast_quantile` object.
+#' @param quantile_level The desired quantile level for the point forecast.
+#' Defaults to 0.5 (median).
```

Comment:
Surprised to see this as an argument - do you have a use case in mind where one would want it not to be 0.5?

Reply:
Not reallly but I thought there was little issue with giving this option

Reply:
Agree that it's a minor issue but it does add to the cognitive burden on anyone coming tot he function so if we think there's no use case it might be better to simplify?